src/test/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/WarningTaskWorkerHandlerTest.java

```diff
@@ -87,17 +112,35 @@ void should_complete_warning_external_task_Service() {
 
             verify(externalTaskService).complete(externalTask, expectedProcessVariables);
             verify(taskManagementServiceApi).addTaskNote(S2S_TOKEN, externalTask.getId(), getExpectedWarningRequest());
+            verify(camundaClient).updateProcessVariables(
+                S2S_TOKEN, PROCESS_INSTANCE_ID,
+                getAddProcessVariableRequest(expectedWarningValues)
+            );
+        }
 
+        @NotNull
+        private AddProcessVariableRequest getAddProcessVariableRequest(String expectedWarningValues) {
+            Map<String, DmnValue<String>> modifications = new HashMap<>();
+            try {
+                WarningValues warningValues = new WarningValues(expectedWarningValues);
+
+                String warning = warningValues.getValuesAsJson();
+                modifications.put("warningList", DmnValue.dmnStringValue(warning));
+            } catch (JsonProcessingException e) {
+                // do nothing
+            }
+            return new AddProcessVariableRequest(modifications);
         }
 
         @Test
         void should_complete_warning_external_task_Service_with_duplicate_warnings() {
```

Comment:
If possible, can we test with arbitrary values. For eg: warningCode is different, but text is same.